tests/hare/assert_hare.py

```diff
@@ -51,23 +54,69 @@ def assert_all(curr_idx, ns):
     total = testconfig['bootstrap']['replicas'] + testconfig['client']['replicas']
 
     # assert no node ended up with an empty set at the end of the pre-round
-    lst = query_pre_round(curr_idx, ns)
+    lst = query_pre_round(curr_idx, ns, 1)
     assert 0 == len(lst)
 
     # assert all nodes had SVP ready at the end of round 1
-    lst = query_round_1(curr_idx, ns)
+    lst = query_round_1(curr_idx, ns, 1)
     assert total == len(lst)
 
     # assert all nodes had a (non-nil) proposal at the end of round 2
-    lst = query_round_2(curr_idx, ns)
+    lst = query_round_2(curr_idx, ns, 1)
     assert total == len(lst)
 
     # assert at least f+1 has committed at the end of round 3
-    lst = query_round_3(curr_idx, ns)
+    lst = query_round_3(curr_idx, ns, 1)
     f = int(testconfig['client']['args']['hare-max-adversaries'])
     assert len(lst) >= f + 1
 
     # assert termination output set
-    lst = query_hare_output_set(curr_idx, ns)
+    lst = query_hare_output_set(curr_idx, ns, 1)
     assert total == len(lst)
     assert validate(lst)
+
+
+def expect_consensus_process(curr_idx, ns, layer):
+    total = testconfig['bootstrap']['replicas'] + testconfig['client']['replicas']
```

Comment:
since these assert are basically not related to a specific running test, receiving `total` as a parameter lets you you this function offline wihout any tests running

Reply:
This function is currently not in use  but I fixed it anyway